### PR TITLE
Implement initial Welder server extension

### PR DESCRIPTION
**Pull request by calbach**
Still needs tests.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [x] Get a thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: docker/jupyter/custom/jupyter_delocalize.py [hunk 1]
```diff
@@ -158,3 +159,61 @@ def delete_file(self, path):
...
+
+  def _welder_delocalize(self, path):
+    # Ignore storage link failure, throw other errors.
+    resp = requests.post('http://localhost:8080/objects', data=json.dumps({
+      'action': 'safeDelocalize',
+      # Sometimes the Jupyter UI provided "path" contains a leading /, sometimes
+      # not; strip for Welder.
+      'localPath': path.lstrip('/')
+    }))
+    if resp.status_code != requests.codes.ok:
```
Comment (Qi77Qi) [inline docker/jupyter/custom/jupyter_delocalize.py]: does `requests.code.ok` include 204? I recently changed return code to 204...The fact it was 200 previously was an oversight

Comment (calbach) [reply docker/jupyter/custom/jupyter_delocalize.py]: Good catch, done.

File: docker/jupyter/custom/jupyter_delocalize.py [hunk 2]
```diff
@@ -158,3 +159,61 @@ def delete_file(self, path):
...
+  def __init__(self, *args, **kwargs):
+    # This log line shouldn't be necessary, but Jupyter's built-in logging is
+    # lacking and its configuration can be complex. Having this in the server
+    # logs is useful for confirming which ContentsManager is in use.
+    self.log.info('initializing WelderContentsManager')
+    super(WelderContentsManager, self).__init__(*args, **kwargs)
+
+  def _welder_delocalize(self, path):
+    # Ignore storage link failure, throw other errors.
+    resp = requests.post('http://localhost:8080/objects', data=json.dumps({
```
Comment (rtitle) [inline docker/jupyter/custom/jupyter_delocalize.py]: Does `localhost` work from inside the notebook container? I think I remember having issues with localhost and had to use `127.0.0.1`. Could be wrong though.

**Comment by rtitle** [comment]
Looks fine to me. jenkins retest for good measure.

File: docker/jupyter/custom/jupyter_delocalize.py (hunk 2 above)
Comment (calbach) [reply docker/jupyter/custom/jupyter_delocalize.py]: It works when I test it locally with docker-compose; I think network_mode: host should allow it. Switched to `127.0.0.1` for at least consistency with cluster-site.conf; verified this still works with docker-compose as well.

**Comment by calbach** [comment]
Added some tests, which turned up a few bugs. This should be good now.

**Comment by calbach** [comment]
jenkins retest (?)

**Pull request merged**